### PR TITLE
Peer JSON serialization/deserialization

### PR DESCRIPTION
**Pull request by georgysavva**
# Description of change

This PR adds JSON serialization/deserialization methods to `Peer` struct. The original need for this is the ability to manually pass neighbors to the gossip layer via JSON HTTP API or configuration file in Goshimmer, see https://github.com/iotaledger/goshimmer/pull/1191.

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Enhancement

## How the change has been tested

The code is covered with unittests 

## Change checklist

Add an `x` to the boxes that are relevant to your changes, and delete any items that are not.

- [x] My code follows the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: autopeering/peer/peer.go [hunk 1]
```diff
@@ -126,3 +129,37 @@ func Unmarshal(data []byte) (*Peer, error) {
...
 	return FromProto(s)
 }
+
+type peerJSON struct {
+	PublicKey string          `json:"publicKey"`
+	IP        net.IP          `json:"ip"`
+	Services  *service.Record `json:"services"`
+}
+
+// UnmarshalJSON de-serializes given json data into a Peer struct.
```
Comment (luca-moser) [inline autopeering/peer/peer.go:139]:
```suggestion
// UnmarshalJSON deserializes given json data into a Peer struct.
```

just for consistency as you wrote `deserializes` at `func (s *Record) UnmarshalJSON`

File: crypto/ed25519/public_key.go [hunk 2]
```diff
@@ -12,6 +14,16 @@ import (
 // PublicKey is the type of Ed25519 public keys.
 type PublicKey [PublicKeySize]byte
 
+// PublicKeyFromString parses give string with base58 encoding and returns a PublicKey.
```
Comment (luca-moser) [inline crypto/ed25519/public_key.go:17]:
```suggestion
// PublicKeyFromString parses the given string with base58 encoding and returns a PublicKey.
```

**Review by luca-moser — APPROVED** [review]
Just two remarks re some func comments. Otherwise lgtm

File: autopeering/peer/peer.go (hunk 1 above)
Comment (georgysavva) [reply autopeering/peer/peer.go]: I have no idea why I wrote it like that here :) Thanks for fixing!

**Review by capossele — APPROVED** [review]
lgtm

File: crypto/ed25519/public_key.go [hunk 3]
```diff
@@ -12,6 +14,16 @@ import (
 // PublicKey is the type of Ed25519 public keys.
 type PublicKey [PublicKeySize]byte
 
+// PublicKeyFromString parses the given string with base58 encoding and returns a PublicKey.
+func PublicKeyFromString(s string) (publicKey PublicKey, err error) {
```
[the remark this answers is not in the document]
Comment (georgysavva) [reply crypto/ed25519/public_key.go:18]: I named it like that for symmetry with the existing opposite method: `.String()` that also uses base58 encoding. 
I think if in the future we have more encodings we should come up with better names for both methods.
So I would keep `PublicKeyFromString` for now.

**Pull request merged**